types/src/get_with_proof.rs

```diff
@@ -357,7 +361,12 @@ fn verify_get_events_by_access_path_resp(
     events_with_proof: &[EventWithProof],
     proof_of_latest_event: &AccountStateWithProof,
 ) -> Result<()> {
-    let account_resource = get_account_resource_or_default(&proof_of_latest_event.blob)?;
+    let account_resource: AccountResource = TryFrom::<&AccountStateBlob>::try_from(
+        proof_of_latest_event
+            .blob
+            .as_ref()
+            .ok_or_else(|| format_err!("Missing proof_of_latest_event.blob"))?,
+    )?;
```

Comment:
Same here. If the request asks for an account that doesn't exist, it would be valid for the server to return `None`.

Reply:
Similar question here; should we just return early with `Ok()` if `proof_of_latest_event.blob` is `None`?

Reply:
This one is similar -- no matter it's `Some` or `None` we need to do `proof_of_latest_event.verify` to determine whether the server is honest. Then this info is combined with other stuff to determine if the events make sense.